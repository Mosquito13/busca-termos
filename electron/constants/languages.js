module.exports = {
  USA: 'en-us',
  BRAZIL: 'pt-br',
  SPAIN: 'es',
  PORTUGAL: 'pt',
  GERMANY: 'de',
  FRANCE: 'fr',
  ROMANIA: 'ro',
  ITALIA: 'it',
  SLOVAKIA: 'sk',
  TURKEY: 'tr',
  RUSSIA: 'ru',
  POLAND: 'pl',
  CATALONIA: 'ca',
  DENMARK: 'da',
  FINLAND: 'fi',
  CZECH: 'cs'
};
