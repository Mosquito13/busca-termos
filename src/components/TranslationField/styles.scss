@import '../../sass/styles.scss';

.translation-field {
    display: flex;

    &__flag {
        flex: 0 0 4rem;
        display: flex;
        align-items: center;
        justify-content: center;
        @include themify() {
            background-color: themed('color-base-darker');
            border-right: 1px solid themed('color-border');
        }
    }

    &__icon {
        filter: drop-shadow(0 0 3px $color-black) opacity(.8);
    }
}