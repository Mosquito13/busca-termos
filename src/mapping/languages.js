import { ReactComponent as BrazilIcon } from '../assets/svg/brazil.svg';
import { ReactComponent as CataloniaIcon } from '../assets/svg/catalonia.svg';
import { ReactComponent as CzechIcon } from '../assets/svg/czech.svg';
import { ReactComponent as DenmarkIcon } from '../assets/svg/denmark.svg';
import { ReactComponent as FinlandIcon } from '../assets/svg/finland.svg';
import { ReactComponent as FranceIcon } from '../assets/svg/france.svg';
import { ReactComponent as GermanyIcon } from '../assets/svg/germany.svg';
import { ReactComponent as ItalyIcon } from '../assets/svg/italy.svg';
import { ReactComponent as PolandIcon } from '../assets/svg/poland.svg';
import { ReactComponent as PortugalIcon } from '../assets/svg/portugal.svg';
import { ReactComponent as RomaniaIcon } from '../assets/svg/romania.svg';
import { ReactComponent as RussiaIcon } from '../assets/svg/russia.svg';
import { ReactComponent as SlovakiaIcon } from '../assets/svg/slovakia.svg';
import { ReactComponent as SpainIcon } from '../assets/svg/spain.svg';
import { ReactComponent as TurkeyIcon } from '../assets/svg/turkey.svg';
import { ReactComponent as UsaIcon } from '../assets/svg/usa.svg';

const languageMapping = {
  BRAZIL: {
    id: 'pt-br',
    title: 'Português (Brasil)',
    getIcon: () => <BrazilIcon />
  },
  CATALONIA: {
    id: 'ca',
    title: 'Catalão',
    getIcon: () => <CataloniaIcon />
  },
  CZECH: {
    id: 'cs',
    title: 'Tcheco',
    getIcon: () => <CzechIcon />
  },
  DENMARK: {
    id: 'da',
    title: 'Dinamarquês',
    getIcon: () => <DenmarkIcon />
  },
  FINLAND: {
    id: 'fi',
    title: 'Finlandês',
    getIcon: () => <FinlandIcon />
  },
  FRANCE: {
    id: 'fr',
    title: 'Francês',
    getIcon: () => <FranceIcon />
  },
  GERMANY: {
    id: 'de',
    title: 'Alemão',
    getIcon: () => <GermanyIcon />
  },
  ITALY: {
    id: 'it',
    title: 'Italiano',
    getIcon: () => <ItalyIcon />
  },
  POLAND: {
    id: 'pl',
    title: 'Polonês',
    getIcon: () => <PolandIcon />
  },
  PORTUGAL: {
    id: 'pt',
    title: 'Português (Portugal)',
    getIcon: () => <PortugalIcon />
  },
  ROMANIA: {
    id: 'ro',
    title: 'Romeno',
    getIcon: () => <RomaniaIcon />
  },
  RUSSIA: {
    id: 'ru',
    title: 'Russo',
    getIcon: () => <RussiaIcon />
  },
  SLOVAKIA: {
    id: 'sk',
    title: 'Eslovaco',
    getIcon: () => <SlovakiaIcon />
  },
  SPAIN: {
    id: 'es',
    title: 'Espanhol',
    getIcon: () => <SpainIcon />
  },
  TURKEY: {
    id: 'tr',
    title: 'Turco',
    getIcon: () => <TurkeyIcon />
  },
  USA: {
    id: 'en-us',
    title: 'Inglês (EUA)',
    getIcon: () => <UsaIcon />
  }
};

export { languageMapping };
