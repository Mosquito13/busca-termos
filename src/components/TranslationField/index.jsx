import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import Icon, { SMALL } from '../common/Icon';
import InputText from '../common/InputText';

import coreSelectors from '../../selectors/core';
import { loadTranslation } from '../../actions/translation';
import translationSelectors from '../../selectors/translation';

import './styles.scss';

const TranslationField = ({ id, tooltip, icon }) => {
  const dispatch = useDispatch();
  const translation = useSelector((state) => translationSelectors.getTranslation(state, id));
  const selectedTermId = useSelector(coreSelectors.getSelectedItemId);

  useEffect(() => {
    if (selectedTermId) {
      dispatch(loadTranslation(id, selectedTermId));
    }
  }, [dispatch, id, selectedTermId]);

  return (
    <div className="translation-field">
      <div className="translation-field__flag">
        <div className="translation-field__icon" title={tooltip}>
          <Icon size={SMALL}>{icon}</Icon>
        </div>
      </div>
      <InputText
        placeholder="Selecione um item na lista"
        value={translation}
        borderless
        disabled
      />
    </div>
  );
};

TranslationField.propTypes = {
  /**
   * Identificador do idioma
   */
  id: PropTypes.string,
  /**
   * Hint da bandeira do campo
   */
  tooltip: PropTypes.string,
  /**
   * Icone exibido ao lado do campo
   */
  icon: PropTypes.node
};

export default TranslationField;
