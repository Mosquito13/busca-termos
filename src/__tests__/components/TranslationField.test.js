import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';

import TranslationField from '../../components/TranslationField';
import { languageMapping } from '../../mapping/languages';

const getStateMock = (selectedItemId) => ({
  Core: {
    selectedItemId
  },
  Translation: {
    [languageMapping.USA.id]: 'Translated',
    [languageMapping.TURKEY.id]: null
  }
});

const prepareReactReduxMocks = (selectedItemId, dispatchMock) => {
  jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => dispatchMock);
  jest
    .spyOn(reactRedux, 'useSelector')
    .mockImplementation((selector) => selector(getStateMock(selectedItemId)));
};

describe(`${TranslationField.name}`, () => {
  describe('tests', () => {
    let wrapper, dispatchMock;

    beforeAll(() => {
      dispatchMock = jest.fn();

      prepareReactReduxMocks(null, dispatchMock);

      wrapper = mount(
        <TranslationField
          id={languageMapping.USA.id}
          tooltip={languageMapping.USA.title}
          icon={languageMapping.USA.getIcon()}
        />
      );
    });

    it('should render and not fire loadTranslation when selection is empty', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(dispatchMock).toBeCalledTimes(0);
    });

    it('should fire loadTranslation when has selection', () => {
      wrapper.unmount();

      prepareReactReduxMocks(321, dispatchMock);

      wrapper = mount(
        <TranslationField
          id={languageMapping.TURKEY.id}
          tooltip={languageMapping.TURKEY.title}
          icon={languageMapping.TURKEY.getIcon()}
        />
      );

      expect(dispatchMock).toBeCalledTimes(1);
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render', () => {
      prepareReactReduxMocks(null, jest.fn());

      const tree = renderer
        .create(
          <TranslationField
            id={languageMapping.TURKEY.id}
            tooltip={languageMapping.TURKEY.title}
            icon={languageMapping.TURKEY.getIcon()}
          />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
