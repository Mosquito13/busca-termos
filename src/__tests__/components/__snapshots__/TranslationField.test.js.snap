// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationField snapshots should render 1`] = `
<div
  className="translation-field"
>
  <div
    className="translation-field__flag"
  >
    <div
      className="translation-field__icon"
      title="Turco"
    >
      <div
        className="icon icon--small"
      >
        <svg>
          turkey.svg
        </svg>
      </div>
    </div>
  </div>
  <div
    className="input-text"
  >
    <input
      className="input-text__input input-text__input--disabled input-text__input--borderless"
      disabled={true}
      onChange={[Function]}
      placeholder="Selecione um item na lista"
      spellCheck={false}
      type="text"
      value=""
    />
    <div
      className="input-text__copy-button"
    >
      <div
        className="copy-text-button"
        onClick={[Function]}
        title="Copiar"
      >
        <div
          className="icon icon--small"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="13"
              rx="2"
              ry="2"
              width="13"
              x="9"
              y="9"
            />
            <path
              d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
